lightkurve/lightcurve.py

```diff
@@ -1224,3 +1226,114 @@ def box(amplitude, depth, to, width):
+    maximizes the posterior probability of the fit.
+
+    Parameters
+    ----------
+    lc : LightCurve object
+        An object from KeplerLightCurve or LightCurve.
+        Note that flattening the lightcurve beforehand does aid the quest
+        for the transit period.
+    min_period : float
+        Minimum period to search for. Units must be the same as `lc.time`.
+    max_period : float
+        Maximum period to search for. Units must be the same as `lc.time`.
+    nperiods : int
+        Number of periods to search between `min_period` and `max_period`.
+    period_scale : str
+        Type of the scale used to create the grid of periods between `min_period`
+        and `max_period` used to search for the best period.
+        Options are 'linear' or 'log'.
+
+    Returns
+    -------
+    log_posterior : list
+        Log posterior (up to an additive constant) of the fit. The "best"
+        period is therefore the one that maximizes the log posterior
+        probability.
+    trial_periods : numpy array
+        List of trial periods.
+    best_period : float
+        Best period.
+
+    Notes
+    -----
+    This function is experimental. Changes may be made in both its signature
+    and implementation.
+    """
+
+    t = np.linspace(-.5, .5, len(lc.time))
+    logprior_to = oktopus.prior.UniformPrior(lb=-.5, ub=.5)
+    logprior_width = oktopus.prior.UniformPrior(lb=1e-3, ub=.2)
+    m = np.nanmean(lc.flux)
+
+    def logposterior(args, amplitude=None, depth=None, data=None):
+        to, width = args
+        out_of_transit = (t < to) + (t >= to + width)
+        in_transit = np.logical_not(out_of_transit)
+        ll = ((data - amplitude) ** 2 * out_of_transit
+              + (data - (amplitude - depth)) ** 2 * in_transit)
+        return np.nansum(ll) + logprior_to(to) + logprior_width(width)
+
+    def opt_amplitude(width, depth):
+        return width * depth + m
+
+    def opt_depth(to, width, data):
+        in_transit = (t < to + width) * (t >= to)
+        n = np.nanmean(data[in_transit])
+        return (m - n) / (1 - width)
+
+    if period_scale == 'linear':
+        trial_periods = np.linspace(min_period, max_period, nperiods)
+    elif period_scale == 'log':
+        trial_periods = np.logspace(np.log10(min_period), np.log10(max_period), nperiods)
+    elif period_scale == 'inverse':
+        trial_periods = 1/np.linspace(1/max_period, 1/min_period, nperiods)
+    else:
+        raise ValueError("period_scale must be one of {}. Got {}."
+                         .format("{'linear', 'log'}", period_scale))
+
+    log_posterior = []
+    for p in tqdm(trial_periods):
+        folded = lc.fold(period=p)
+        amplitude_star, depth_star = m, 1e-4
+        width_star = .1
+        if np.nanmean(folded.flux[t < 0]) > np.nanmean(folded.flux[t > 0]):
+            to_star = .25
+        else:
+            to_star = -.25
+        for i in range(niters):
```

Comment:
Do you want to add a few lines of comments inside these for loops, just to guide the untrained eye where in the algorithm we are?